src/backend/core/api/viewsets/message_template.py

```diff
@@ -0,0 +1,185 @@
+
+    def get_queryset(self):
+        """Get queryset filtered by user permissions and custom filters."""
+        # For detail actions (retrieve, update, delete), use different logic
+        if self.action in ["retrieve", "update", "partial_update", "destroy"]:
+            return self.get_queryset_for_detail()
+
+        return self.get_queryset_for_list()
+
+    def get_queryset_for_detail(self):
+        """Get queryset for detail actions (retrieve, update, delete)."""
+        user = self.request.user
+
+        if user.is_superuser:
+            return MessageTemplate.objects.all()
+
+        # Filter by user permissions for regular users
+        accessible_mailboxes = Mailbox.objects.filter(accesses__user=user)
+        accessible_maildomains = MailDomain.objects.filter(accesses__user=user)
+
+        return MessageTemplate.objects.filter(
+            Q(template_mailboxes__mailbox__in=accessible_mailboxes)
+            | Q(template_maildomains__maildomain__in=accessible_maildomains)
+        ).distinct()
+
+    def get_queryset_for_list(self):
+        """Get queryset for list action with filtering."""
+        user = self.request.user
+
+        # At least one of mailbox_id or maildomain_id is required
+        mailbox_id = self.request.query_params.get("mailbox_id")
+        maildomain_id = self.request.query_params.get("maildomain_id")
+        if not mailbox_id and not maildomain_id:
+            return MessageTemplate.objects.none()
+
+        # filter by user permissions
+        if user.is_superuser:
+            queryset = MessageTemplate.objects.all()
+        elif mailbox_id:
+            accessible_mailboxes = Mailbox.objects.filter(accesses__user=user)
+            queryset = MessageTemplate.objects.filter(
+                template_mailboxes__mailbox_id__in=accessible_mailboxes
+            )
+        elif maildomain_id:
+            accessible_maildomains = MailDomain.objects.filter(accesses__user=user)
+            queryset = MessageTemplate.objects.filter(
+                template_maildomains__maildomain_id__in=accessible_maildomains
+            )
+
+        is_default = self.request.query_params.get("is_default")
+        # filter by mailbox_id or maildomain_id
+        if mailbox_id:
+            queryset = queryset.filter(template_mailboxes__mailbox_id=mailbox_id)
+            if is_default is not None:
+                is_default_bool = is_default.lower() in ("true", "1", "yes")
+                queryset = queryset.filter(
+                    template_mailboxes__is_default=is_default_bool
+                )
+        if maildomain_id:
+            queryset = queryset.filter(
+                template_maildomains__maildomain_id=maildomain_id
+            )
+            if is_default is not None:
+                is_default_bool = is_default.lower() in ("true", "1", "yes")
+                queryset = queryset.filter(
+                    template_maildomains__is_default=is_default_bool
+                )
```

Comment:
When the user specifies a `mailbox_id`, `MessageTemplate` linked to the maildomain's mailbox should be also returned.